src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -541,7 +541,14 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         ASR::array_index_t ai;
         ai.loc = x.base.base.loc;
         ai.m_left = nullptr;
-        ai.m_right = index;
+        // ASR uses 1-based indexing
```

Comment:
Actually, ASR does not use 1 based indexing. Rather, you have to explicitly specify lower and upper bound.